app/services/indexer.rb

```diff
@@ -86,12 +92,28 @@ def workflow_client
     @workflow_client ||= WorkflowClientFactory.build
   end
 
-  def dor_services_client
-    @dor_services_client ||= DorServicesClientFactory.build
+  def cocina_repository
+    lambda do |druid|
+      Dor::Services::Client.object(druid).find
+    rescue Dor::Services::Client::Error => e
+      raise DorIndexing::RepositoryError, e.message
+    end
   end
 
-  def cocina_repository
-    @cocina_repository ||= CocinaRepository.new
+  def administrative_tags_repository
+    lambda do |druid|
+      Dor::Services::Client.object(druid).administrative_tags.list
+    rescue Dor::Services::Client::Error => e
+      raise DorIndexing::RepositoryError, e.message
+    end
+  end
+
+  def release_tags_repository
+    lambda do |druid|
+      Dor::Services::Client.object(druid).release_tags.list
+    rescue Dor::Services::Client::Error => e
+      raise DorIndexing::RepositoryError, e.message
+    end
```

Comment:
I'll add test coverage back for when these callables fail.

Reply:
@justinlittman I looked into adding test coverage here, but these lambdas are invoked within the dor_indexing gem, not with the app, so to test this in DIA would involve testing the internals of the gem. I could spend time doing that but it seems like something we wouldn't ordinarily do. Advise? If you agree it's beyond the scope of this PR and/or not worth the time, feel free to approve/merge?